src/main/java/org/prebid/server/analytics/reporter/greenbids/GreenbidsAnalyticsReporter.java

```diff
@@ -170,8 +203,57 @@ private GreenbidsPrebidExt toGreenbidsPrebidExt(ObjectNode adapterNode) {
         }
     }
 
-    private boolean isNotEmptyObjectNode(JsonNode analytics) {
-        return analytics != null && analytics.isObject() && !analytics.isEmpty();
+    private Map<String, Ortb2ImpExtResult> extractAnalyticsResultFromAnalyticsTag(BidResponse bidResponse) {
+        return Optional.ofNullable(bidResponse)
+                .map(BidResponse::getExt)
+                .map(ExtBidResponse::getPrebid)
+                .map(ExtBidResponsePrebid::getModules)
+                .map(ExtModules::getTrace)
+                .map(ExtModulesTrace::getStages)
```

Comment:
Trace log will be available only in debug mode, maybe it is better to use `AuctionContext.hookExecutionContext`?

Reply:
Replaced extraction of analytics tags from bid respone to AuctionContext.hookExecutionContext